plugin/storage/memory/memory.go

```diff
@@ -21,28 +21,38 @@ import (
 
 	"github.com/jaegertracing/jaeger/model"
 	"github.com/jaegertracing/jaeger/model/adjuster"
+	"github.com/jaegertracing/jaeger/pkg/memory/config"
 	"github.com/jaegertracing/jaeger/storage/spanstore"
 )
 
 var errTraceNotFound = errors.New("Trace was not found")
 
-// Store is an unbounded in-memory store of traces
+// Store is an in-memory store of traces
 type Store struct {
-	// TODO: make this a bounded memory store
 	sync.RWMutex
+	ids        []*model.TraceID
 	traces     map[model.TraceID]*model.Trace
 	services   map[string]struct{}
 	operations map[string]map[string]struct{}
 	deduper    adjuster.Adjuster
+	config     *config.Configuration
+	index      int
 }
 
-// NewStore creates an in-memory store
+// NewStore creates an unbounded in-memory store
 func NewStore() *Store {
+	return WithConfiguration(&config.Configuration{MaxTraces: 0})
+}
+
+// WithConfiguration creates a new in memory storage based on the given configuration
+func WithConfiguration(configuration *config.Configuration) *Store {
```

Comment:
I've always preferred passing around configurations as values instead of references. The caller could technically update the config which might lead to errant behavior

Reply:
And storing it in the parent struct by value as well